### PR TITLE
Cutting release 2.1.4

### PR DESCRIPTION
**Pull request by keithboone**
Cutting release 2.1.4

### CODE REVIEW

File: pom.xml
Comment (keithboone) [inline pom.xml:1]: Release build, upgrading BC to 2.0

File: src/main/java/gov/cdc/izgateway/common/HealthService.java [hunk 1]
```diff
@@ -37,4 +37,12 @@ public static void setServerName(String serverName) {
 	public static void setBuildName(String build) {
 		health.setBuildName(build);
 	}
+
```
Comment (keithboone) [inline src/main/java/gov/cdc/izgateway/common/HealthService.java:40]: Adjust setter to deal with multiple dbs for future 2.2 migration case.

File: src/main/java/gov/cdc/izgateway/configuration/AppProperties.java [hunk 2]
```diff
@@ -21,6 +21,11 @@ public class AppProperties {
 	@Getter
 	@Value("${server.mode:prod}")
 	private String serverMode;
+	
+	@Getter
```
Comment (keithboone) [inline src/main/java/gov/cdc/izgateway/configuration/AppProperties.java:25]: Prep for different db types.

File: src/main/java/gov/cdc/izgateway/logging/event/Health.java [hunk 3]
```diff
@@ -85,7 +85,11 @@ public class Health {
 	@JsonProperty 
 	@Schema(description="Host name as known by the operating system")
 	private String hostname;
-
+
```
Comment (keithboone) [inline src/main/java/gov/cdc/izgateway/logging/event/Health.java:88]: Report the database.

File: src/main/java/gov/cdc/izgateway/soap/MockMessage.java [hunk 4]
```diff
@@ -153,9 +153,14 @@ public enum MockMessage {
 			HttpStatus.BAD_REQUEST),
 
 	TC_24H(MediaType.APPLICATION_XML, MockMessageText.TC_24H_TEXT,
-			HttpStatus.BAD_REQUEST), TC_UNKF(MockMessage::simulateFault,
+			HttpStatus.BAD_REQUEST),
```
Comment (keithboone) [inline src/main/java/gov/cdc/izgateway/soap/MockMessage.java:156]: Added test case for invalid XML

File: src/main/java/gov/cdc/izgateway/soap/SoapControllerBase.java [hunk 5]
```diff
@@ -580,6 +580,7 @@ protected ResponseEntity<FaultMessage> handleFault(Fault fault) {
 		logFault(fault);
 		FaultMessage faultMessage = new FaultMessage(fault, messageNamespace);
 		faultMessage.updateAction(isHub());
+		logResponseMessage(faultMessage);
```
Comment (keithboone) [inline src/main/java/gov/cdc/izgateway/soap/SoapControllerBase.java:583]: Fix for IGDD-1738

File: src/main/java/gov/cdc/izgateway/soap/fault/HubClientFault.java [hunk 6]
```diff
@@ -136,14 +136,28 @@ private HubClientFault(MessageSupport messageSupport, IDestination destination,
...
+	 * @param dest	The destination
+	 * @param statusCode	The status code of the response
+	 * @param body	The message body
+	 * @return
+	 */
+	public static HubClientFault invalidMessage(Throwable rootCause, IDestination dest, int statusCode, InputStream body) {
 		String bodyString = XmlUtils.toString(body);
 		if (statusCode != 500 && statusCode != 200) {
-			return new HubClientFault(getHttpMessageSupport(statusCode), dest, rootCause, statusCode, bodyString, result);
+			return new HubClientFault(getHttpMessageSupport(statusCode), dest, rootCause, statusCode, bodyString, null);
```
Comment (keithboone) [inline src/main/java/gov/cdc/izgateway/soap/fault/HubClientFault.java:150]: More fixing IGDD-1738

**Review by pcahillai — APPROVED** [review]
The changes look good.

**Pull request closed without merging**